src/live_restore/live_restore_fs.c

```diff
@@ -1207,109 +1071,105 @@ __wt_live_restore_fh_import_extents_from_string(
+        __live_restore_fh_free_bitmap(session, lr_fh);
     }
-    if (__wt_rwlock_islocked(session, &lr_fh->ext_lock))
-        __wt_readunlock(session, &lr_fh->ext_lock);
+    __wt_readunlock(session, &lr_fh->bitmap_lock);
     return (ret);
 }
 
 /*
- * __wt_live_restore_fh_extent_to_metadata --
- *     Given a WiredTiger file handle generate a string of its extents. If live restore is not
- *     running or the extent list is missing, which indicates the file is complete, return a
- *     WT_NOTFOUND error.
+ * __live_restore_encode_bitmap --
+ *     Encode a live restore bitmap as a hexadecimal string. The caller must free the bitmap string.
+ */
+static int
+__live_restore_encode_bitmap(
+  WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh, WT_ITEM *buf)
+{
+    WT_ASSERT_ALWAYS(session, __wt_rwlock_islocked(session, &lr_fh->bitmap_lock),
+      "Live restore lock not taken when needed");
+    if (lr_fh->destination.bitmap_size == 0 || lr_fh->destination.complete == true)
+        return (0);
+    size_t bitmap_byte_count = lr_fh->destination.bitmap_size / 8;
+    if (lr_fh->destination.bitmap_size % 8 != 0)
+        bitmap_byte_count++;
+    return (__wt_raw_to_hex(session, lr_fh->destination.bitmap, bitmap_byte_count, buf));
+}
+
+/*
+ * __wt_live_restore_fh_to_metadata --
+ *     Given a WiredTiger file handle generate a metadata string. If live restore is not running
+ *     return a WT_NOTFOUND error.
  */
 int
-__wt_live_restore_fh_extent_to_metadata(
-  WT_SESSION_IMPL *session, WT_FILE_HANDLE *fh, WT_ITEM *extent_string)
+__wt_live_restore_fh_to_metadata(WT_SESSION_IMPL *session, WT_FILE_HANDLE *fh, WT_ITEM *meta_string)
 {
+    WT_DECL_RET;
     if (!F_ISSET(S2C(session), WT_CONN_LIVE_RESTORE_FS))
         return (WT_NOTFOUND);
 
     WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh = (WTI_LIVE_RESTORE_FILE_HANDLE *)fh;
-    if (lr_fh->destination.complete)
-        return (WT_NOTFOUND);
 
-    wt_off_t prev_off = 0;
-    WTI_LIVE_RESTORE_HOLE_NODE *head = lr_fh->destination.hole_list_head;
-    WT_RET(__wt_buf_catfmt(session, extent_string, ",live_restore="));
-    while (head != NULL) {
-        WT_RET(__wt_buf_catfmt(
-          session, extent_string, "%" PRId64 "-%" WT_SIZET_FMT, head->off - prev_off, head->len));
-        prev_off = head->off;
-        if (head->next != NULL)
-            WT_RET(__wt_buf_catfmt(session, extent_string, ";"));
-        head = head->next;
-    }
-    __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE,
-      "Appending live restore extents (%s) to metadata for file handle %s", fh->name,
-      (char *)extent_string->data);
+    WT_ITEM buf;
+    WT_CLEAR(buf);
 
-    return (0);
+    __wt_readlock(session, &lr_fh->bitmap_lock);
+
+    WT_ERR(__live_restore_encode_bitmap(session, lr_fh, &buf));
+    WT_ERR(
+      __wt_buf_catfmt(session, meta_string, ",live_restore=(bitmap=%s,bitmap_size=%" PRIu64 ")",
+        buf.size == 0 ? "" : (char *)buf.data, lr_fh->destination.bitmap_size));
```

Comment:
Typically an empty config in the metadata is `live_restore=`. I.e. not including an empty string. Did you consider that? It could look like:
if (lr_fh->destination.bitmap_size > 0)
    wt_buf_catfmt(...",live_restore=(bitmap...
else
    wt_buf_catfmt(...",live_restore=");

Reply:
805689feb